pkg/controller/prpqr_reconciler/prpqr_reconciler.go

```diff
@@ -60,14 +60,19 @@ type injectingResolverClient interface {
 }
 
 type prowConfigGetter interface {
-	Config() periodicDefaulter
+	Defaulter() periodicDefaulter
+	Config() *prowconfig.Config
 }
 
 type wrappedProwConfigAgent struct {
 	pc *prowconfig.Agent
 }
 
-func (w *wrappedProwConfigAgent) Config() periodicDefaulter {
+func (w *wrappedProwConfigAgent) Defaulter() periodicDefaulter {
+	return w.pc.Config()
+}
+
+func (w *wrappedProwConfigAgent) Config() *prowconfig.Config {
```

Comment:
Could you explain this to me?

Reply:
Sure, I originally needed a `prowconfig.Config` in this func:
```golang
func generateAggregatedProwjobs(uid string, ciopConfig *api.ReleaseBuildConfiguration, defaulter periodicDefaulter, baseCiop *api.Metadata, prpqrName, prpqrNamespace string, spec *v1.ReleaseJobSpec, prs []v1.PullRequestUnderTest, inject *api.MetadataWithTest) ([]*prowv1.ProwJob, error)
```
and I noticed the interface `defaulter periodicDefaulter` was backed by:
```golang
type wrappedProwConfigAgent struct {
  pc *prowconfig.Agent
}
```
which holds the config I want, since it can be retrieved with `pc.Config()`.
Therefore instead of adding another param `cfg *prowconfig.Config` to that already long func signature, like so:
```diff
- func generateAggregatedProwjobs(uid string, ciopConfig *api.ReleaseBuildConfiguration, defaulter periodicDefaulter, baseCiop *api.Metadata, prpqrName, prpqrNamespace string, spec *v1.ReleaseJobSpec, prs []v1.PullRequestUnderTest, inject *api.MetadataWithTest) ([]*prowv1.ProwJob, error)
+ func generateAggregatedProwjobs(uid string, ciopConfig *api.ReleaseBuildConfiguration, defaulter periodicDefaulter, baseCiop *api.Metadata, prpqrName, prpqrNamespace string, spec *v1.ReleaseJobSpec, prs []v1.PullRequestUnderTest, inject *api.MetadataWithTest, cfg *prowconfig.Config) ([]*prowv1.ProwJob, error)
```
I've extended the interface definition:
```diff
type prowConfigGetter interface {
  Defaulter() periodicDefaulter
+ Config() *prowconfig.Config
}
```
and then replaced `defaulter periodicDefaulter` with `getCfg prowConfigGetter` as in this way I can either obtain a `getCfg.Defaulter()` or a `getCfg.Config()`.